pkg/runtime/pubsub/default_bulkpub_test.go

```diff
@@ -0,0 +1,149 @@
+*/
+
+package pubsub
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	contribPubsub "github.com/dapr/components-contrib/pubsub"
+	daprt "github.com/dapr/dapr/pkg/testing"
+)
+
+func TestBulkPublish_DefaultBulkPublisher(t *testing.T) {
+	req := &contribPubsub.BulkPublishRequest{
+		Entries: []contribPubsub.BulkMessageEntry{
+			{
+				EntryId:     "78a48b5c-ff5a-4275-9bef-4a3bb8eefc3b",
+				Event:       []byte("event1"),
+				ContentType: "text/plain",
+				Metadata:    map[string]string{},
+			},
+			{
+				EntryId:     "d64669e2-fab6-4452-a933-8de44e26ca02",
+				Event:       []byte("event2"),
+				ContentType: "text/plain",
+				Metadata:    map[string]string{},
+			},
+		},
+		PubsubName: "pubsub",
+		Topic:      "topic",
+		Metadata:   map[string]string{},
+	}
+
+	tcs := []struct {
+		name                   string
+		bulkPublishSeriallyKey string
+		publishErrors          []error
+		expectError            bool
+	}{
+		{
+			name:                   "bulkPublishSeriallyKey is set to true, without publish errors",
+			bulkPublishSeriallyKey: "true",
+			publishErrors:          []error{nil, nil},
+			expectError:            false,
+		},
+		{
+			name:                   "bulkPublishSeriallyKey is set to true, with a publish error",
+			bulkPublishSeriallyKey: "true",
+			publishErrors:          []error{errors.New("publish error"), nil},
+			expectError:            true,
+		},
+		{
+			name:                   "bulkPublishSeriallyKey is not true, without publish errors",
+			bulkPublishSeriallyKey: "false",
+			publishErrors:          []error{nil, nil},
+			expectError:            false,
+		},
+		{
+			name:                   "bulkPublishSeriallyKey is not true, with a publish errors",
+			bulkPublishSeriallyKey: "false",
+			publishErrors:          []error{nil, errors.New("publish error")},
+			expectError:            true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(fmt.Sprintf("publishes all messages in a request, %s", tc.name), func(t *testing.T) {
+			// Create publish requests for each message in the bulk request.
+			var pubReqs []*contribPubsub.PublishRequest
+			for _, entry := range req.Entries {
+				contenType := entry.ContentType
+				pubReqs = append(pubReqs, &contribPubsub.PublishRequest{
+					Data:        entry.Event,
+					ContentType: &contenType,
+					Metadata:    entry.Metadata,
+					PubsubName:  req.PubsubName,
+					Topic:       req.Topic,
+				})
+			}
+
+			// Set up the mock pubsub to return the publish errors.
+			mockPubSub := &daprt.MockPubSub{Mock: mock.Mock{}}
+			for i, e := range tc.publishErrors {
+				mockPubSub.Mock.On("Publish", pubReqs[i]).Return(e)
+			}
+			bulkPublisher := NewDefaultBulkPublisher(mockPubSub)
+
+			// Set up metadata and invoke the bulk publish method.
+			req.Metadata[bulkPublishSeriallyKey] = tc.bulkPublishSeriallyKey
+			res, err := bulkPublisher.BulkPublish(context.TODO(), req)
```

Comment:
`context.Background()` because this is not a TODO

Reply:
Addressed in #5396